File: src/main/java/Main/ItemClasses/Armor.java
package Main.ItemClasses;

public class Armor extends Item{
    private Integer armorValue;
    private ArmorClasses armorClass;

    public Armor() {
        super();
        this.standardIconName = "armorIcon.png";
    }

    public Armor(
            String name,
            String description,
            int weight,
            int valueInGold,
            int amount,
            int armorValue,
            ArmorClasses armorClass
    ) {
        super(name, description, weight, valueInGold, amount);
        this.armorValue = armorValue;
        this.armorClass = armorClass;
        this.standardIconName = "armorIcon.png";
    }

    public Integer getArmorValue() {
        return armorValue;
    }

    public void setArmorValue(Integer armorValue) {
        this.armorValue = armorValue;
    }

    public ArmorClasses getArmorClass() {
        return this.armorClass;
    }

    public void setArmorClass(ArmorClasses armorClass) {
        this.armorClass = armorClass;
    }
}

File: src/main/kotlin/main/ItemClasses/Armor.kt
package main.ItemClasses

import kotlinx.serialization.Serializable

@Serializable
class Armor() : Item() {
    init {
        standardIconName = "armorIcon.png"
    }
    
    var armorValue: Int = 10
    var armorClass: ArmorClasses = ArmorClasses.MEDIUM

    constructor(name: String, description: String, weight: Int, valueInGold: Int, amount: Int, armorValue: Int, armorClass: ArmorClasses) : this() {
        this.name = name
        this.description = description
        this.weight = weight
        this.valueInGold = valueInGold
        this.amount = amount
        this.armorValue = armorValue
        this.armorClass = armorClass
    }
}
